### ISSUE TITLE
[HOLD for GH-4362 merge] PLP page in general in terms of viewMode, state not same when going back

### ISSUE THREAD

**Issue by bgambocjaviniar** [SAP/spartacus#5143]
When the user is in grid mode and chooses a product, and go back, the state should still be the same as what it was before.

According to the PWA checklist, the state should be the same when clicking back. Need to start after GH-4362 is merged

https://developers.google.com/web/progressive-web-apps/checklist#pressing-back-from-a-detail-page-retains-scroll-position-on-the----------previous-list-page

## Steps to Reproduce
- go to a plp page
- change to grid view
- click any product
- click back from the browser

## Observed Results

it changes back to list view

## Expected Results
it needs to stay as grid view

**Comment by tobi-or-not-tobi** [comment]
I believe this must be part of a bigger story, as the number of issue pile up.
In my view, this is part of the UI state, which contains the grid/list view, scroll behaviour (specifically infinitely loading data when scrolling), the facet selection, etc. 

- we do not store the UI state and therefor cannot resolve it internally when navigating the UI. 
- we we do not store the UI state in the URL and will therefor not be able to bookmark or share the state
- we do not use an easy UI pattern to navigate back (specifically when we go from pdp back to plp, it' currently very difficult to maintain the exact state - we should consider "virtual scrolling" for this.
- the way we persist the search facets is ugly and not user friendly

In other words: can we start an epic around this and approach it with some more groundwork before trying to solve things ad-hoc?

FYI @Xymmer

**Comment by giancorderoortiz** [comment]
Still an issue in Spartacus 3.0. Fixing for 4.0 ? 

![Screen Shot 2021-06-06 at 11 29 13 AM](https://user-images.githubusercontent.com/46171897/120930272-8efb9080-c6ba-11eb-9a9b-a46e2295cdbb.png)

![Screen Shot 2021-06-06 at 11 29 20 AM](https://user-images.githubusercontent.com/46171897/120930273-902cbd80-c6ba-11eb-935c-52298097db78.png)

Back to PLP
![Screen Shot 2021-06-06 at 11 29 26 AM](https://user-images.githubusercontent.com/46171897/120930274-902cbd80-c6ba-11eb-8487-2e0b9811f7b4.png)
